noobaa_sa/s3_client.py

```diff
@@ -312,3 +307,83 @@ def delete_all_objects_in_bucket(self, bucket_name):
         # TODO: Support buckets with versioning enabled
         log.info(f"Deleting all objects in bucket {bucket_name} via boto3")
         self._boto3_resource.Bucket(bucket_name).objects.all().delete()
+
+    def generate_objects_for_upload(
+        self,
+        amount=1,
+        min_size="1M",
+        max_size="1M",
+        files_dir="",
+    ):
+        """
+        Generate random objects for upload purpose
+
+        Args:
+            amount (int): The number of objects to write
+            min_size(str): The minimum size of each object,
+                        specified in a format understood by the 'dd' command.
+            max_size(str): The maximum size of each object,
+                        specified in a format understood by the 'dd' command.
+            files_dir (str): A directory where the objects will be written
+                        locally. If not specified, a temporary directory will
+                        be used.
+
+        Returns:
+            list: A list of the names of the objects written to the bucket
+
+        """
+        # Determine the directory to use for file storage
+        actual_files_dir = files_dir if files_dir else tempfile.mkdtemp()
+        written_objs = []
+        # Generate random files in the specified directory
+        written_objs += generate_random_files(
+            actual_files_dir, amount, min_size, max_size
+        )
+        return written_objs
```

Comment:
I might be just missing it, but why use this instead of `common_ci_utils.random_utils.generate_random_files`?

Reply:
Fixed